### ISSUE TITLE
Editor: keyboard input for font mapres

### ISSUE THREAD

**Issue by ChillerDragon** [ddnet/ddnet#3570]
Im 'corneum suggested to add keyboard typing support for font mapres (https://github.com/chillerbot/chillerbot-ux/issues/12)

I quickly hacked it together and its somewhat working in my custom client.

https://zillyhuhn.com/OpenTube/video.php?t=chillerbot_ux_editor_text.mp4&u=github

@Learath2 suggested to pr it against ddnet so I decided to open this issue first to discuss if this is wanted.
Its basic and there are still some issues:

- [x] no number support
- [ ] no special character support (like ?!.,:-_)
- [x] hardcodet alphabet index (not all fonts start with index 1 being a)
- [ ] no proper keyboard layout support (fixed qwerty layout so german keyboards for example have z and y swapped)
- [ ] no filter, so it works on all tile and even game layers (feature or bug?)
- [x] no text cursor
- [ ] bad UX (non obvious workflow of pressing T and clicking a tile to start)
- [ ] performance issues when zoomed out

Is it worth to pr it in this state? Can be tested in [chillerbot-ux](https://github.com/chillerbot/chillerbot-ux) its not as bad as the name suggests.

**Comment by Learath2** [comment]
I like the idea, maybe if we standardize a layout for text mapres this could work?

**Comment by ChillerDragon** [comment]
> I like the idea, maybe if we standardize a layout for text mapres this could work?

Yes I think having "A" at index "1" could be a sane standard that helps. But some fonts have numbers and some do not and the position is not always the same due to special characters.

I suggest picking a sane standard. Probably based on the mapres included in ddnet client already and use it as default.
Then add support for optional mapres_name.json file that holds some metadata.

**Comment by Robyt3** [comment]
Fixed by #8331.
